modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultConstantFoldingOptimizationPhase.java

```diff
@@ -832,6 +832,18 @@ public void visitInvokeCallMember(InvokeCallMemberNode irInvokeCallMemberNode, C
         }
     }
 
+    private String convertIntToOrdinal(int i) {
+        String[] suffixes = new String[] { "th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th" };
+        switch (i % 100) {
+            case 11:
+            case 12:
+            case 13:
+                return i + "th";
+            default:
+                return i + suffixes[i % 10];
+        }
+    }
```

Comment:
I'm not sure we can just lift code from https://stackoverflow.com/questions/6810336/is-there-a-way-in-java-to-convert-an-integer-to-its-ordinal-name/6810409#6810409 - it's CC-BY-SA and we're dual SSPL/Elastic v2. I don't know.